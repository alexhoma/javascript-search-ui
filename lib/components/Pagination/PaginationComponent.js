"use strict";
var __extends = (this && this.__extends) || (function () {
    var extendStatics = function (d, b) {
        extendStatics = Object.setPrototypeOf ||
            ({ __proto__: [] } instanceof Array && function (d, b) { d.__proto__ = b; }) ||
            function (d, b) { for (var p in b) if (Object.prototype.hasOwnProperty.call(b, p)) d[p] = b[p]; };
        return extendStatics(d, b);
    };
    return function (d, b) {
        if (typeof b !== "function" && b !== null)
            throw new TypeError("Class extends value " + String(b) + " is not a constructor or null");
        extendStatics(d, b);
        function __() { this.constructor = d; }
        d.prototype = b === null ? Object.create(b) : (__.prototype = b.prototype, new __());
    };
})();
exports.__esModule = true;
var preact_1 = require("preact");
var Template_1 = require("../Template");
var Helpers_1 = require("./Helpers");
var NavigationComponent_1 = require("./NavigationComponent");
var PaginationActions_1 = require("./PaginationActions");
/**
 * Pagination Component
 */
var PaginationComponent = /** @class */ (function (_super) {
    __extends(PaginationComponent, _super);
    /**
     * Constructor
     */
    function PaginationComponent() {
        var _this = _super.call(this) || this;
        /**
         * Handle click
         *
         * @param page
         */
        _this.handleClick = function (page) {
            var _a = _this.props, store = _a.store, environmentId = _a.environmentId, repository = _a.repository;
            var currentResult = store.getCurrentResult();
            var currentQuery = store.getCurrentQuery();
            var totalPages = (0, Helpers_1.getTotalPages)(currentResult.getTotalHits(), currentQuery.getSize());
            /**
             * Do not let go further
             */
            if (page <= 0)
                page = 1;
            if (page >= totalPages)
                page = totalPages;
            if (currentQuery.getPage() === page) {
                return;
            }
            /**
             * Dispatch change page action
             */
            (0, PaginationActions_1.paginationChangeAction)(environmentId, currentQuery, repository, page);
        };
        _this.state = {
            page: 1
        };
        return _this;
    }
    /**
     * Component will receive props
     *
     * @param props
     */
    PaginationComponent.prototype.componentWillReceiveProps = function (props) {
        var page = props.store.getCurrentQuery().getPage();
        this.setState(function (prevState) {
            return {
                page: page
            };
        });
    };
    /**
     * Render
     *
     * @return {any}
     */
    PaginationComponent.prototype.render = function () {
        var _this = this;
        var props = this.props;
        var currentResult = props.store.getCurrentResult();
        if (props.store.getCurrentResult() == null) {
            return;
        }
        var currentQuerySize = props.store.getCurrentQuery().getSize();
        var totalPages = (0, Helpers_1.getTotalPages)(currentResult.getTotalHits(), currentQuerySize);
        /**
         * Hide container if hits are empty
         */
        if (currentResult.getTotalHits() === 0 ||
            totalPages === 1) {
            return null;
        }
        var padding = props.padding;
        var goFirstLast = props.goFirstLast;
        var containerClassName = props.classNames.container;
        var itemClassName = props.classNames.item;
        var activeClassName = props.classNames.active;
        var disabledClassName = props.classNames.disabled;
        var nextClassName = props.classNames.next;
        var previousClassName = props.classNames.previous;
        var lastClassName = props.classNames.last;
        var firstClassName = props.classNames.first;
        var itemTemplate = props.template.item;
        var nextTemplate = props.template.next;
        var previousTemplate = props.template.previous;
        var firstTemplate = props.template.first;
        var lastTemplate = props.template.last;
        var currentQueryPage = props.store.getCurrentQuery().getPage();
        /**
         * Get Total pages
         */
        var pages = (0, Helpers_1.totalPagesToArray)(totalPages);
        /**
         *  Get pages spectre
         */
        var spectreSize = (padding * 2) + 1;
        var isTouchingLeft = currentQueryPage <= (padding + 1);
        var isTouchingRight = (currentQueryPage + padding) >= totalPages;
        var spectre = pages.slice((0, Helpers_1.getStart)(totalPages, padding, currentQueryPage, spectreSize, isTouchingLeft, isTouchingRight), (0, Helpers_1.getEnd)(totalPages, padding, currentQueryPage, spectreSize, isTouchingLeft, isTouchingRight));
        /**
         * Dynamic disabled classes
         */
        var previousDisabledClass = (currentQueryPage === 1) ? disabledClassName : '';
        var nextDisabledClass = (currentQueryPage === totalPages) ? disabledClassName : '';
        return ((0, preact_1.h)("ul", { className: "as-pagination " + containerClassName },
            (0, preact_1.h)(NavigationComponent_1["default"], { isVisible: goFirstLast, classNames: "as-pagination__item as-pagination__item--first " + firstClassName + " " + previousDisabledClass, template: firstTemplate, handleClick: function () { return _this.handleClick(1); } }),
            (0, preact_1.h)(NavigationComponent_1["default"], { isVisible: true, classNames: "as-pagination__item as-pagination__item--previous " + previousClassName + " " + previousDisabledClass, template: previousTemplate, handleClick: function () { return _this.handleClick(currentQueryPage - 1); } }),
            spectre.map(function (page) { return ((0, preact_1.h)("li", { className: "as-pagination__item as-pagination__item--link " + itemClassName + " " + ((currentQueryPage === page) ? activeClassName : ''), onClick: function () { return _this.handleClick(page); } },
                (0, preact_1.h)(Template_1["default"], { template: itemTemplate, data: { page: page.toLocaleString('de-DE') }, dictionary: _this.props.dictionary }))); }),
            (0, preact_1.h)(NavigationComponent_1["default"], { isVisible: true, classNames: "as-pagination__item as-pagination__item--next " + nextClassName + " " + nextDisabledClass, template: nextTemplate, handleClick: function () { return _this.handleClick(currentQueryPage + 1); } }),
            (0, preact_1.h)(NavigationComponent_1["default"], { isVisible: goFirstLast, classNames: "as-pagination__item as-pagination__item--last " + lastClassName + " " + nextDisabledClass, template: lastTemplate, handleClick: function () { return _this.handleClick(totalPages); } })));
    };
    return PaginationComponent;
}(preact_1.Component));
PaginationComponent.defaultProps = {
    padding: 3,
    goFirstLast: false,
    classNames: {
        container: '',
        item: '',
        active: 'as-pagination__item--active',
        disabled: 'as-pagination__item--disabled',
        next: '',
        first: '',
        previous: '',
        last: ''
    },
    template: {
        item: '{{page}}',
        next: '>',
        previous: '<',
        first: '<<',
        last: '>>'
    }
};
exports["default"] = PaginationComponent;
