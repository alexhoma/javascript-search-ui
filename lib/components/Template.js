"use strict";
var __extends = (this && this.__extends) || (function () {
    var extendStatics = function (d, b) {
        extendStatics = Object.setPrototypeOf ||
            ({ __proto__: [] } instanceof Array && function (d, b) { d.__proto__ = b; }) ||
            function (d, b) { for (var p in b) if (Object.prototype.hasOwnProperty.call(b, p)) d[p] = b[p]; };
        return extendStatics(d, b);
    };
    return function (d, b) {
        extendStatics(d, b);
        function __() { this.constructor = d; }
        d.prototype = b === null ? Object.create(b) : (__.prototype = b.prototype, new __());
    };
})();
exports.__esModule = true;
var preact_1 = require("preact");
var Hogan = require("hogan.js");
/**
 * Template
 */
var Template = /** @class */ (function (_super) {
    __extends(Template, _super);
    function Template() {
        var _this = _super !== null && _super.apply(this, arguments) || this;
        /**
         * Render template
         *
         * @param template
         * @param result
         *
         * @return {any}
         */
        _this.renderTemplate = function (template, result) {
            /**
             * Compile template using hogan.js
             */
            var compiledTemplate = Hogan.compile(template);
            var output = compiledTemplate.render(result);
            return {
                __html: output
            };
        };
        return _this;
    }
    /**
     * Render
     *
     * @return {any}
     */
    Template.prototype.render = function () {
        var props = this.props;
        var template = props.template;
        var data = props.data;
        var className = props.className;
        return (template)
            ? preact_1.h("div", { className: className, dangerouslySetInnerHTML: this.renderTemplate(template, data) })
            : null;
    };
    return Template;
}(preact_1.Component));
exports["default"] = Template;
