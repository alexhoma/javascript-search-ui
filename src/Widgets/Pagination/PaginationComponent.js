/**
 * @jsx h
 */
import { h, Component } from 'preact';
import {paginationChangeAction} from "./paginationActions";

/**
 * Pagination Component
 */
class PaginationComponent extends Component {
    constructor() {
        super();
        this.state = {
            currentPage: 1,
            shownPages: 4
        }
    }

    handleClick = (page) => {
        const {
            environmentId,
            currentQuery,
            client
        } = this.props;

        /**
         * Dispatch change page action
         */
        paginationChangeAction(
            {
                selectedPage: page
            },
            {
                environmentId,
                currentQuery,
                client
            }
        );
    };

    componentWillReceiveProps(props) {
        const { page } = props.data.query;
        this.setState({
            currentPage: page ? page : 1
        });
    }

    render() {
        const {
            classNames: {
                container: containerClassName
            },
            currentQuery,
            data
        } = this.props;

        let itemsPerPage = parseInt(currentQuery.size);
        let totalHits = parseInt(data.total_hits);

        let totalPages = Math.ceil(totalHits / itemsPerPage);
        let pages = totalPagesToArray(totalPages);

        // -----------------------------------

        let {shownPages, currentPage} = this.state;
        const isTouchingLeft = currentPage <= Math.round(shownPages / 2);
        const isTouchingRight = (currentPage + Math.round(shownPages / 2)) > totalPages;


        // pages spectre
        const start = () => {
            if (isTouchingLeft) {
                return currentPage - (currentPage % shownPages);
            }
            return currentPage - Math.round(shownPages / 2);
        };

        const end = () => {
            if (isTouchingLeft) {
                return (shownPages / currentPage) * currentPage;
            }
            if(isTouchingRight) {
                return (currentPage / shownPages) * (currentPage / shownPages)
            }
            return (currentPage) + Math.round(shownPages / 2);
        };

        let pagesSpectre = pages.slice(
            start(),
            end()
        );

        return (
            <div className={`asui-pagination pagination-list ${containerClassName}`}>
                <span
                    className={`pagination-link`}
                    onClick={() => this.handleClick(this.state.currentPage - 1)}
                >
                    Prev
                </span>
                {pagesSpectre.map(page => {
                    return (
                        <span
                            className={`pagination-link ${(this.state.currentPage === page) ? 'is-current' : ''}`}
                            onClick={() => this.handleClick(page)}
                        >
                            {page}
                        </span>
                    )
                })}

                <span
                    className={`pagination-link`}
                    onClick={() => this.handleClick(this.state.currentPage + 1)}
                >
                    Next
                </span>
            </div>
        );
    }
}



function totalPagesToArray(totalPages) {
    let pages = [];
    for(let index = 1; index < totalPages; index++) {
        pages.push(index);
    }

    return pages;
}

PaginationComponent.defaultProps = {
    classNames: {
        container: ''
    }
};

export default PaginationComponent;