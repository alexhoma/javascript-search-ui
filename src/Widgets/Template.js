/** @jsx h */
import { h, Component } from 'preact';
import Hogan from 'hogan.js';

class Template extends Component {
    renderTemplate = (template, data) => {
        let compiledTemplate = Hogan.compile(template);
        let output = compiledTemplate.render(data);


        return {
            __html: output
        };
    };

    render() {
        const {
            template,
            data,
            className
        } = this.props;

        return (template)
            ? <div
                className={className}
                dangerouslySetInnerHTML={this.renderTemplate(template, data)}
            />
            : null
        ;
    }
}

export default Template;