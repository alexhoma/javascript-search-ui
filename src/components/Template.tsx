import { h, Component } from 'preact';
import {TemplateProps} from "./TemplateProps";
import * as Mustache from 'mustache';

/**
 * Template
 */
class Template extends Component<TemplateProps> {

    /**
     * Render template
     *
     * @param template
     * @param result
     * @param dictionary
     *
     * @return {any}
     */
    renderTemplate = (template, result, dictionary) => {

        let trans = function() {
            return function(text, render) {
                return dictionary[text] ?? text;
            }
        }

        let output = Mustache.render(template, {...result, ...{
            'trans': trans
        }});

        return {
            __html: output
        };
    };

    /**
     * Render
     *
     * @return {any}
     */
    render() {

        const props = this.props;
        const template = props.template;
        const data = props.data;
        const className = props.className;
        const dictionary = props.dictionary ?? {};

        return (template)
            ? <div
                className={className}
                dangerouslySetInnerHTML={this.renderTemplate(template, data, dictionary)}
            />
            : null
        ;
    }
}

export default Template;
