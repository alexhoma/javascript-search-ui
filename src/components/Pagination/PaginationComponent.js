/**
 * @jsx h
 */
import { h, Component } from 'preact';
import Template from "../Template";
import {paginationChangeAction} from "./paginationActions";
import NavigationComponent from "./NavigationComponent";
import {
    getTotalPages,
    totalPagesToArray,
    getStart,
    getEnd
} from "./helpers";

/**
 * Pagination Component
 */
class PaginationComponent extends Component {

    handleClick = (page) => {
        const {
            data,
            environmentId,
            currentQuery,
            client
        } = this.props;

        let totalPages = getTotalPages({
            totalHits: data.total_hits,
            hitsPerPage: currentQuery.size
        });

        /**
         * Do not let go further
         */
        if (page <= 0) page = 1;
        if (page >= totalPages) page = totalPages;

        /**
         * Dispatch change page action
         */
        paginationChangeAction(
            {
                selectedPage: page
            },
            {
                environmentId,
                currentQuery,
                client
            }
        );
    };

    render() {
        const {
            padding,
            goFirstLast,
            classNames: {
                container: containerClassName,
                item: itemClassName,
                active: activeClassName,
                disabled: disabledClassName,
                next: nextClassName,
                previous: previousClassName,
                last: lastClassName,
                first: firstClassName,
            },
            template: {
                item: itemTemplate,
                next: nextTemplate,
                previous: previousTemplate,
                first: firstTemplate,
                last: lastTemplate
            },
            currentQuery: {
                page: currentQueryPage,
                size: currentQuerySize
            },
            data
        } = this.props;

        /**
         * Get Total pages
         */
        let totalPages = getTotalPages({
            totalHits: data.total_hits,
            hitsPerPage: currentQuerySize
        });
        let pages = totalPagesToArray(totalPages);

        /**
         *  Get pages spectre
         */
        const paginationSettings = {
            totalPages,
            padding,
            currentPage: currentQueryPage,
            spectreSize: (padding * 2) + 1,
            isTouchingLeft: currentQueryPage <= (padding + 1),
            isTouchingRight: (currentQueryPage + padding) >= totalPages
        };
        let spectre = pages.slice(
            getStart(paginationSettings),
            getEnd(paginationSettings)
        );

        /**
         * Dynamic disabled classes
         */
        let previousDisabledClass = (currentQueryPage === 1) ? disabledClassName : '';
        let nextDisabledClass = (currentQueryPage === totalPages) ? disabledClassName : '';

        /**
         * Hide container if hits are empty
         */
        if (data.total_hits === 0) return null;

        return (
            <ul className={`as-pagination ${containerClassName}`}>
                <NavigationComponent
                    isVisible={goFirstLast}
                    classNames={`as-pagination__item as-pagination__item--first ${firstClassName} ${previousDisabledClass}`}
                    template={firstTemplate}
                    handleClick={() => this.handleClick(1)}
                />
                <NavigationComponent
                    isVisible={true}
                    classNames={`as-pagination__item as-pagination__item--previous ${previousClassName} ${previousDisabledClass}`}
                    template={previousTemplate}
                    handleClick={() => this.handleClick(currentQueryPage - 1)}
                />

                {spectre.map(page => (
                    <li
                        className={`as-pagination__item as-pagination__item--link ${itemClassName} ${
                            (currentQueryPage === page) ? activeClassName : ''
                        }`}
                        onClick={() => this.handleClick(page)}
                    >
                        <Template
                            template={itemTemplate}
                            data={{page: parseInt(page).toLocaleString('de-DE')}}
                        />
                    </li>
                ))}

                <NavigationComponent
                    isVisible={true}
                    classNames={`as-pagination__item as-pagination__item--next ${nextClassName} ${nextDisabledClass}`}
                    template={nextTemplate}
                    handleClick={() => this.handleClick(currentQueryPage + 1)}
                />
                <NavigationComponent
                    isVisible={goFirstLast}
                    classNames={`as-pagination__item as-pagination__item--last ${lastClassName} ${nextDisabledClass}`}
                    template={lastTemplate}
                    handleClick={() => this.handleClick(totalPages)}
                />
            </ul>
        );
    }
}

PaginationComponent.defaultProps = {
    padding: 3,
    goFirstLast: false,
    classNames: {
        container: '',
        item: '',
        active: 'as-pagination__item--active',
        disabled: 'as-pagination__item--disabled',
        next: '',
        first: '',
        previous: '',
        last: ''
    },
    template: {
        item: '{{page}}',
        next: '>',
        previous: '<',
        first: '<<',
        last: '>>'
    }
};

export default PaginationComponent;