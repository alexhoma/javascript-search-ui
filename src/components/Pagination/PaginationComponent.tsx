import { h, Component } from 'preact';
import Template from "../Template";
import {paginationChangeAction} from "./PaginationActions";
import {PaginationState} from "./PaginationState";
import NavigationComponent from "./NavigationComponent";
import {
    getTotalPages,
    totalPagesToArray,
    getStart,
    getEnd
} from "./Helpers";
import {PaginationProps} from "./PaginationProps";

/**
 * Pagination Component
 */
class PaginationComponent extends Component<PaginationProps, PaginationState> {

    /**
     * Constructor
     */
    constructor() {
        super();
        this.state = {
            page: 1
        }
    }

    /**
     * Component will receive props
     *
     * @param props
     */
    componentWillReceiveProps(props) {

        const page = props.currentQuery.getPage();
        this.setState(prevState => {
            return {
                page: page
            };
        });
    }

    /**
     * Handle click
     *
     * @param page
     */
    handleClick = (page) => {
        const {
            currentResult,
            environmentId,
            currentQuery,
            repository
        } = this.props;

        let totalPages = getTotalPages(
            currentResult.getTotalHits(),
            currentQuery.getSize()
        );

        /**
         * Do not let go further
         */
        if (page <= 0) page = 1;
        if (page >= totalPages) page = totalPages;

        if (currentQuery.getPage() === page) {
            return;
        }

        /**
         * Dispatch change page action
         */
        paginationChangeAction(
            environmentId,
            currentQuery,
            repository,
            page
        );
    };

    /**
     * Render
     *
     * @return {any}
     */
    render() {

        const props = this.props;
        const currentResult = props.currentResult;
        if (props.currentResult == null) {
            return;
        }

        const currentQuerySize = props.currentQuery.getSize();
        const totalPages = getTotalPages(
            currentResult.getTotalHits(),
            currentQuerySize
        );

        /**
         * Hide container if hits are empty
         */
        if (
            currentResult.getTotalHits() === 0 ||
            totalPages === 1
        ) {
            return null;
        }

        const padding = props.padding;
        const goFirstLast = props.goFirstLast;
        const containerClassName = props.classNames.container;
        const itemClassName = props.classNames.item;
        const activeClassName = props.classNames.active;
        const disabledClassName = props.classNames.disabled;
        const nextClassName = props.classNames.next;
        const previousClassName = props.classNames.previous;
        const lastClassName = props.classNames.last;
        const firstClassName = props.classNames.first;

        const itemTemplate = props.template.item;
        const nextTemplate = props.template.next;
        const previousTemplate = props.template.previous;
        const firstTemplate = props.template.first;
        const lastTemplate = props.template.last;

        const currentQueryPage = props.currentQuery.getPage();

        /**
         * Get Total pages
         */

        const pages = totalPagesToArray(totalPages);

        /**
         *  Get pages spectre
         */
        const spectreSize =(padding * 2) + 1;
        const isTouchingLeft = currentQueryPage <= (padding + 1);
        const isTouchingRight = (currentQueryPage + padding) >= totalPages;

        const spectre = pages.slice(
            getStart(
                totalPages,
                padding,
                currentQueryPage,
                spectreSize,
                isTouchingLeft,
                isTouchingRight
            ),
            getEnd(
                totalPages,
                padding,
                currentQueryPage,
                spectreSize,
                isTouchingLeft,
                isTouchingRight
            )
        );

        /**
         * Dynamic disabled classes
         */
        const previousDisabledClass = (currentQueryPage === 1) ? disabledClassName : '';
        const nextDisabledClass = (currentQueryPage === totalPages) ? disabledClassName : '';

        return (
            <ul className={`as-pagination ${containerClassName}`}>
                <NavigationComponent
                    isVisible={goFirstLast}
                    classNames={`as-pagination__item as-pagination__item--first ${firstClassName} ${previousDisabledClass}`}
                    template={firstTemplate}
                    handleClick={() => this.handleClick(1)}
                />
                <NavigationComponent
                    isVisible={true}
                    classNames={`as-pagination__item as-pagination__item--previous ${previousClassName} ${previousDisabledClass}`}
                    template={previousTemplate}
                    handleClick={() => this.handleClick(currentQueryPage - 1)}
                />

                {spectre.map(page => (
                    <li
                        className={`as-pagination__item as-pagination__item--link ${itemClassName} ${
                            (currentQueryPage === page) ? activeClassName : ''
                        }`}
                        onClick={() => this.handleClick(page)}
                    >
                        <Template
                            template={itemTemplate}
                            data={{page: page.toLocaleString('de-DE')}}
                            dictionary={this.props.dictionary}
                        />
                    </li>
                ))}

                <NavigationComponent
                    isVisible={true}
                    classNames={`as-pagination__item as-pagination__item--next ${nextClassName} ${nextDisabledClass}`}
                    template={nextTemplate}
                    handleClick={() => this.handleClick(currentQueryPage + 1)}
                />
                <NavigationComponent
                    isVisible={goFirstLast}
                    classNames={`as-pagination__item as-pagination__item--last ${lastClassName} ${nextDisabledClass}`}
                    template={lastTemplate}
                    handleClick={() => this.handleClick(totalPages)}
                />
            </ul>
        );
    }
}

PaginationComponent.defaultProps = {
    padding: 3,
    goFirstLast: false,
    classNames: {
        container: '',
        item: '',
        active: 'as-pagination__item--active',
        disabled: 'as-pagination__item--disabled',
        next: '',
        first: '',
        previous: '',
        last: ''
    },
    template: {
        item: '{{page}}',
        next: '>',
        previous: '<',
        first: '<<',
        last: '>>'
    }
};

export default PaginationComponent;
