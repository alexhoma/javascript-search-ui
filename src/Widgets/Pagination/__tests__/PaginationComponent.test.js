/**
 * @jsx h
 */
import {h} from 'preact';
import {deep} from 'preact-render-spy';
import PaginationComponent from './../PaginationComponent';
import Template from '../../Template';

describe('<PaginationComponent />',() => {
    const defaultProps = {
        target: '#some-target',
        currentQuery: {
            page: 1,
            size: 10
        }
    };
    const customProps = {
        padding: 2,
        goFirstLast: true,
        classNames: {
            container: 'custom-container-class',
            item: 'custom-item-class',
            active: 'custom-active-class',
            disabled: 'custom-disabled-class',
            next: 'custom-next-class',
            first: 'custom-first-class',
            previous: 'custom-previous-class',
            last: 'custom-last-class'
        },
        template: {
            item: '{{page}}',
            next: 'Next -->',
            previous: '<-- Prev',
            first: '<| FIRST',
            last: 'LAST |>'
        }
    };
    const data = {
        total_hits: 1000
    };
    const emptyData = {
        total_hits: 0
    };

    it('should render with default props', () => {
        const tree = deep(
            <PaginationComponent
                {...defaultProps}
                data={data}
            />
        );

        /**
         * Testing the default props tree
         */
        expect(tree.component().props).toMatchSnapshot();
        /**
         * Testing rendered html
         */
        expect(tree).toMatchSnapshot();
    });
    it('should render with custom props', () => {
        const tree = deep(
            <PaginationComponent
                {...defaultProps}
                {...customProps}
                data={data}
            />
        );

        /**
         * Testing the default props tree
         */
        expect(tree.component().props).toMatchSnapshot();
        /**
         * Testing rendered html
         */
        expect(tree).toMatchSnapshot();
    });
    it('should not render if data is empty', () => {
        const tree = deep(
            <PaginationComponent
                {...defaultProps}
                data={emptyData}
            />
        );

        /**
         * Testing rendered html
         */
        expect(tree).toMatchSnapshot();
    });
    it('should handleClick() and and go NEXT', () => {
        const tree = deep(
            <PaginationComponent
                {...defaultProps}
                {...customProps}
                data={data}
            />
        );

        const mock = jest.fn();
        tree.find('.asui-pagination--next')[0].attributes.onClick = mock;
        tree.find('.asui-pagination--next').at(0).simulate('click', 2);

        tree.render(
            <PaginationComponent
                {...defaultProps}
                {...customProps}
                currentQuery={{
                    page: 2, size: 10
                }}
                data={data}
            />
        );

        /**
         * Expect onClick to be called
         */
        expect(mock).toBeCalled();

        /**
         * Testing rendered html
         */
        expect(tree).toMatchSnapshot();
    });
    it('should handleClick() and and go PREV', () => {
        const tree = deep(
            <PaginationComponent
                {...defaultProps}
                {...customProps}
                data={data}
            />
        );

        const mock = jest.fn();
        tree.find('.asui-pagination--previous')[0].attributes.onClick = mock;
        tree.find('.asui-pagination--previous').at(0).simulate('click', 1);

        tree.render(
            <PaginationComponent
                {...defaultProps}
                {...customProps}
                currentQuery={{
                    page: 1, size: 10
                }}
                data={data}
            />
        );

        /**
         * Expect onClick to be called
         */
        expect(mock).toBeCalled();

        /**
         * Testing rendered html
         */
        expect(tree).toMatchSnapshot();
    });
});