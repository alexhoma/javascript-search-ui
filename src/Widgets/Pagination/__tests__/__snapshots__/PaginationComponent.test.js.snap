// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<PaginationComponent /> should handleClick() and and go NEXT 1`] = `
preact-render-spy (1 nodes)
-------
<ul class="asui-pagination custom-container-class">
  <li
    class="asui-pagination--item asui-pagination--first custom-first-class "
    onClick={[Function handleClick]}
  >
    <div>
      <| FIRST
    </div>
  </li>
  <li
    class="asui-pagination--item asui-pagination--previous custom-previous-class "
    onClick={[Function handleClick]}
  >
    <div>
      <-- Prev
    </div>
  </li>
  <li
    class="asui-pagination--item asui-pagination--link custom-item-class "
    onClick={[Function onClick]}
  >
    <div>1</div>
  </li>
  <li
    class="asui-pagination--item asui-pagination--link custom-item-class custom-active-class"
    onClick={[Function onClick]}
  >
    <div>2</div>
  </li>
  <li
    class="asui-pagination--item asui-pagination--link custom-item-class "
    onClick={[Function onClick]}
  >
    <div>3</div>
  </li>
  <li
    class="asui-pagination--item asui-pagination--link custom-item-class "
    onClick={[Function onClick]}
  >
    <div>4</div>
  </li>
  <li
    class="asui-pagination--item asui-pagination--link custom-item-class "
    onClick={[Function onClick]}
  >
    <div>5</div>
  </li>
  <li
    class="asui-pagination--item asui-pagination--next custom-next-class "
    onClick={[Function handleClick]}
  >
    <div>Next --></div>
  </li>
  <li
    class="asui-pagination--item asui-pagination--last custom-last-class "
    onClick={[Function handleClick]}
  >
    <div>LAST |></div>
  </li>
</ul>

`;

exports[`<PaginationComponent /> should handleClick() and and go PREV 1`] = `
preact-render-spy (1 nodes)
-------
<ul class="asui-pagination custom-container-class">
  <li
    class="asui-pagination--item asui-pagination--first custom-first-class custom-disabled-class"
    onClick={[Function handleClick]}
  >
    <div>
      <| FIRST
    </div>
  </li>
  <li
    class="asui-pagination--item asui-pagination--previous custom-previous-class custom-disabled-class"
    onClick={[Function handleClick]}
  >
    <div>
      <-- Prev
    </div>
  </li>
  <li
    class="asui-pagination--item asui-pagination--link custom-item-class custom-active-class"
    onClick={[Function onClick]}
  >
    <div>1</div>
  </li>
  <li
    class="asui-pagination--item asui-pagination--link custom-item-class "
    onClick={[Function onClick]}
  >
    <div>2</div>
  </li>
  <li
    class="asui-pagination--item asui-pagination--link custom-item-class "
    onClick={[Function onClick]}
  >
    <div>3</div>
  </li>
  <li
    class="asui-pagination--item asui-pagination--link custom-item-class "
    onClick={[Function onClick]}
  >
    <div>4</div>
  </li>
  <li
    class="asui-pagination--item asui-pagination--link custom-item-class "
    onClick={[Function onClick]}
  >
    <div>5</div>
  </li>
  <li
    class="asui-pagination--item asui-pagination--next custom-next-class "
    onClick={[Function handleClick]}
  >
    <div>Next --></div>
  </li>
  <li
    class="asui-pagination--item asui-pagination--last custom-last-class "
    onClick={[Function handleClick]}
  >
    <div>LAST |></div>
  </li>
</ul>

`;

exports[`<PaginationComponent /> should not render if data is empty 1`] = `
preact-render-spy (1 nodes)
-------
{null}

`;

exports[`<PaginationComponent /> should render with custom props 1`] = `
Object {
  "children": Array [],
  "classNames": Object {
    "active": "custom-active-class",
    "container": "custom-container-class",
    "disabled": "custom-disabled-class",
    "first": "custom-first-class",
    "item": "custom-item-class",
    "last": "custom-last-class",
    "next": "custom-next-class",
    "previous": "custom-previous-class",
  },
  "currentQuery": Object {
    "page": 1,
    "size": 10,
  },
  "data": Object {
    "total_hits": 1000,
  },
  "goFirstLast": true,
  "padding": 2,
  "target": "#some-target",
  "template": Object {
    "first": "<| FIRST",
    "item": "{{page}}",
    "last": "LAST |>",
    "next": "Next -->",
    "previous": "<-- Prev",
  },
}
`;

exports[`<PaginationComponent /> should render with custom props 2`] = `
preact-render-spy (1 nodes)
-------
<ul class="asui-pagination custom-container-class">
  <li
    class="asui-pagination--item asui-pagination--first custom-first-class custom-disabled-class"
    onClick={[Function handleClick]}
  >
    <div>
      <| FIRST
    </div>
  </li>
  <li
    class="asui-pagination--item asui-pagination--previous custom-previous-class custom-disabled-class"
    onClick={[Function handleClick]}
  >
    <div>
      <-- Prev
    </div>
  </li>
  <li
    class="asui-pagination--item asui-pagination--link custom-item-class custom-active-class"
    onClick={[Function onClick]}
  >
    <div>1</div>
  </li>
  <li
    class="asui-pagination--item asui-pagination--link custom-item-class "
    onClick={[Function onClick]}
  >
    <div>2</div>
  </li>
  <li
    class="asui-pagination--item asui-pagination--link custom-item-class "
    onClick={[Function onClick]}
  >
    <div>3</div>
  </li>
  <li
    class="asui-pagination--item asui-pagination--link custom-item-class "
    onClick={[Function onClick]}
  >
    <div>4</div>
  </li>
  <li
    class="asui-pagination--item asui-pagination--link custom-item-class "
    onClick={[Function onClick]}
  >
    <div>5</div>
  </li>
  <li
    class="asui-pagination--item asui-pagination--next custom-next-class "
    onClick={[Function handleClick]}
  >
    <div>Next --></div>
  </li>
  <li
    class="asui-pagination--item asui-pagination--last custom-last-class "
    onClick={[Function handleClick]}
  >
    <div>LAST |></div>
  </li>
</ul>

`;

exports[`<PaginationComponent /> should render with default props 1`] = `
Object {
  "children": Array [],
  "classNames": Object {
    "active": "active",
    "container": "",
    "disabled": "disabled",
    "first": "",
    "item": "",
    "last": "",
    "next": "",
    "previous": "",
  },
  "currentQuery": Object {
    "page": 1,
    "size": 10,
  },
  "data": Object {
    "total_hits": 1000,
  },
  "goFirstLast": false,
  "padding": 3,
  "target": "#some-target",
  "template": Object {
    "first": "<< First",
    "item": "{{page}}",
    "last": "Last >>",
    "next": "Next >",
    "previous": "< Prev",
  },
}
`;

exports[`<PaginationComponent /> should render with default props 2`] = `
preact-render-spy (1 nodes)
-------
<ul class="asui-pagination ">
  <li
    class="asui-pagination--item asui-pagination--previous  disabled"
    onClick={[Function handleClick]}
  >
    <div>
      < Prev
    </div>
  </li>
  <li
    class="asui-pagination--item asui-pagination--link  active"
    onClick={[Function onClick]}
  >
    <div>1</div>
  </li>
  <li
    class="asui-pagination--item asui-pagination--link  "
    onClick={[Function onClick]}
  >
    <div>2</div>
  </li>
  <li
    class="asui-pagination--item asui-pagination--link  "
    onClick={[Function onClick]}
  >
    <div>3</div>
  </li>
  <li
    class="asui-pagination--item asui-pagination--link  "
    onClick={[Function onClick]}
  >
    <div>4</div>
  </li>
  <li
    class="asui-pagination--item asui-pagination--link  "
    onClick={[Function onClick]}
  >
    <div>5</div>
  </li>
  <li
    class="asui-pagination--item asui-pagination--link  "
    onClick={[Function onClick]}
  >
    <div>6</div>
  </li>
  <li
    class="asui-pagination--item asui-pagination--link  "
    onClick={[Function onClick]}
  >
    <div>7</div>
  </li>
  <li
    class="asui-pagination--item asui-pagination--next  "
    onClick={[Function handleClick]}
  >
    <div>Next ></div>
  </li>
</ul>

`;
